articles/10-tips-for-working-with-node.js/index.md

```diff
@@ -0,0 +1,368 @@
+date: 2020-08-22T00:00:00-10:00	
+topics: [Node.js]	
+excerpt_separator: <!--more-->	
+images:		
+  - url: /engineering-education/10-tips-for-working-with-nodejs/hero.jpg	
+    alt: laptop node.js website
+---
+As a developer getting started with Node.js application development,
+there a few basic skills that they need to be familiar with as they move
+along to build their First Node.js application.
+
+This article will cover some of the common tips you need and practice 
+in order to have a strong background fromgetting started to deploying 
+the first Node.js application.
+
+### 1.  Understand Node.js. Package Manager (Npm)
+
+This is a package manager for JavaScript Node.js Platform that helps in
+package installation, version management, and dependency management,
+which is very easy to use. Npm helps to install the package you want
+with just a single line command. To use Npm you need Node.js to be
+installed in your gadget. For OS X and Window OS, the Node.js installer
+has an easy to use interface and Npm is included with Node.js
+Installation. Developers using Linux need to run the following command
+inapt package managed to set up Node.js and Npm
+
+```
+sudo apt-get update
+
+sudo apt-get install node
+
+sudo apt-get install npm
+```
+
+You can then verify Npm installation by writing the following in your
+terminal `npm -v` or if you realize you have an older version run `npm
+install npm -g` (for windows) or `npm install -g npm\@latest` (for Linus
+and mac) to update to the current Npm version. Npm helps to install
+Node.js modules. For example, you just need to write `npm install`
+followed by your specified module ie 
+`npm install express`, once the
+package is installed you will be able to use `require ()` on your Node.js
+project and they will run just as if they were built-ins. Another
+important aspect of Npm is dependency management. You can run `npm
+install` in your root folder and any listed dependency in your
+package.json (if you have one) will be installed. This help mostly when
+installing Node.js project from git repository For example `vow` (Node.js
+testing framework) can be installed from git and its single dependency
+`eye` can automatically be installed ie
+
+```
+git clone https://github.com/clonehad/vows.git
+
+cd vows
+
+npm install
+```
+
+### 2.  Understand A to Z Most Useful Node.js. Modules and Frameworks
+
+The power of Node.js lies in hundreds of packages that make it easier to
+write your program. The packages are open source and they are the
+ecosystem of Node.js applications. Every Node.js developer will always
+interact with these packages to get a more efficient development
+workflow.
+
+We collect a list of some of the most useful libraries that will help
+you ease the development of Node.js applications.
+
+-   [Express](http://expressjs.com/): it a web application framework for
+    Node.js that handle robust APIs and Web servers
+
+-   [Gulp](https://www.npmjs.com/package/gulp): helps to automate
```

Comment:
Some list items have a full stop at the end and the first word capitialised and some don't. You should be consistent.